### PR TITLE
docs: flesh out tips for typed linting and .eslintrc.cjs

### LINKED ISSUE #6826: Docs: Typed Linting guide causes ESLint complaints in .eslintrc.(c)js
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

When starting a TypeScript repository from scratch using our docs, the typical flow is:

1. [Getting Started](https://typescript-eslint.io/getting-started)
2. [Linting with Type Information](https://typescript-eslint.io/linting/typed-linting)

...however, we don't mention anything about how to get typed linting working (or excluded) for the `.eslintrc.(c)js` file we tell the user to write. As a result, the file receives this lovely complaint:

```plaintext
Parsing error: ESLint was configured to run on `<tsconfigRootDir>/.eslintrc.cjs` using `parserOptions.project`: /users/josh/repos/repros/tsconfig.json
However, that TSConfig does not include this file. Either:
- Change ESLint's list of included files to not include this file
- Change that TSConfig to include this file
- Create a new TSConfig that includes this file and include it in your parserOptions.project
See the typescript-eslint docs for more info: https://typescript-eslint.io/linting/troubleshooting#i-get-errors-telling-me-eslint-was-configured-to-run--however-that-tsconfig-does-not--none-of-those-tsconfigs-include-this-fileeslint
```

Screenshot of the previous lint error in VS Code

That's a very wordy complaint for a first-time typed linting user to receive. Not ideal.

I'm not sure what the best way to resolve this is. A few starting options:

* Use [config `overrides`](https://eslint.org/docs/latest/use/configure/configuration-files#configuration-based-on-glob-patterns) to restrict type checking to `.ts`, `.tsx`, `.cts`, etc. files _(I'm a little negative on this, as it makes the config more complex)_
* Tell them to exclude that ESLint config from being linted _(I don't like this - many of us want to lint all our files)_
* Have the user `allowJs` in their TSConfig used for ESLint _(I really don't like this - users will need to either add a new TSConfig for linting or modify their main project's compiler options)_

...I don't love any of those 🤔. How are users "supposed" to fix this?

This issue came out of a Twitch stream with @Jutanium. 👋

### Affected URL(s)

https://typescript-eslint.io/linting/typed-linting

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

- [x] Addresses an existing open issue: fixes #6826
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Doesn't provide an explicit _"here's the right way to do this!"_ because I don't think there is one 🥲. At least not unless ESLint configs eventually could be written in TypeScript!

Instead, adds an explicit caution note and bolsters the FAQ entry to explicitly handle the idea of a root-level `.cjs`/`.js` file and typed linting.

### CODE REVIEW

File: docs/linting/Troubleshooting.mdx [hunk 1]
```diff
@@ -30,21 +30,67 @@ If you don't find an existing extension rule, or the extension rule doesn't work
 
 ## I get errors telling me "ESLint was configured to run ... However, that TSConfig does not / none of those TSConfigs include this file"
 
+These errors come are caused by an ESLint config requesting type information be generated for a file that isn't included in the TypeScript configuration.
+
 ### Fixing the Error
 
+<!-- prettier-ignore-start -->
```
Comment (JoshuaKGoldberg) [inline docs/linting/Troubleshooting.mdx:37]: Am I doing this wrong? Try removing these disables - is this a bug in Prettier / its mdx parsing? Or is it just me?

File: docs/linting/Troubleshooting.mdx
Comment (JoshuaKGoldberg) [inline docs/linting/Troubleshooting.mdx:1]: @jutanium would love your input on this if you have time 😇

File: docs/linting/Troubleshooting.mdx [hunk 2]
```diff
@@ -30,21 +30,67 @@ If you don't find an existing extension rule, or the extension rule doesn't work
 
 ## I get errors telling me "ESLint was configured to run ... However, that TSConfig does not / none of those TSConfigs include this file"
 
+These errors come are caused by an ESLint config requesting type information be generated for a file that isn't included in the TypeScript configuration.
```
Comment (Josh-Cena) [inline docs/linting/Troubleshooting.mdx:33]:
```suggestion
These errors are caused by an ESLint config requesting type information be generated for a file that isn't included in the TypeScript configuration.
```

File: docs/linting/Troubleshooting.mdx [hunk 3]
```diff
@@ -30,21 +30,67 @@ If you don't find an existing extension rule, or the extension rule doesn't work
...
+              files: ['./**/*.{ts,tsx}'],
+            },
+          ],
+        };
+        ```
+
+      </details>
+      <details>
+        <summary>
+          Alternatively, in our <a href="/blog/announcing-typescript-eslint-v6-beta">version v6</a>, you can add use our{' '}
```
Comment (Josh-Cena) [inline docs/linting/Troubleshooting.mdx:67]:
```suggestion
          Alternatively, in our <a href="/blog/announcing-typescript-eslint-v6-beta">version v6</a>, you can use our{' '}
```

File: docs/linting/Troubleshooting.mdx [hunk 4]
```diff
@@ -30,21 +30,67 @@ If you don't find an existing extension rule, or the extension rule doesn't work
...
   - Use [one of the options ESLint offers](https://eslint.org/docs/latest/user-guide/configuring/ignoring-code) to ignore files, namely a `.eslintignore` file, or `ignorePatterns` config.
 - If you **do** want to lint the file:
   - If you **do not** want to lint the file with [type-aware linting](./Typed_Linting.mdx):
-    - Use [ESLint's `overrides` configuration](https://eslint.org/docs/latest/user-guide/configuring/configuration-files#configuration-based-on-glob-patterns) to configure the file to not be parsed with type information.
-      - A popular setup is to omit the above additions from top-level configuration and only apply them to TypeScript files via an override.
-      - Alternatively, you can add `parserOptions: { project: null }` to an override for the files you wish to exclude. Note that `{ project: undefined }` will not work.
+    - Use [ESLint's `overrides` configuration](https://eslint.org/docs/latest/user-guide/configuring/configuration-files#configuration-based-on-glob-patterns) to configure the file to not be parsed with type information:
+      <details>
+        <summary>
+          A popular setup is remove all typed rules from top-level configuration
```
Comment (Josh-Cena) [inline docs/linting/Troubleshooting.mdx:46]:
```suggestion
          A popular setup is to remove all rules requiring type information from the top-level configuration
```

File: docs/linting/Typed_Linting.mdx [hunk 5]
```diff
@@ -26,10 +26,15 @@ module.exports = {
...
+:::caution
+Your `.eslintrc.cjs` file may start receiving a parsing error about type information.
+See [our TSConfig inclusion FAQ](./Troubleshooting.mdx#i-get-errors-telling-me-eslint-was-configured-to-run--however-that-tsconfig-does-not--none-of-those-tsconfigs-include-this-file).
+:::
+
 In more detail:
 
 - `plugin:@typescript-eslint/recommended-requiring-type-checking` is another [recommended configuration](./CONFIGURATIONS.mdx) we provide. This one contains recommended rules that additionally require type information.
-- `parserOptions.project` tells our parser the relative path where your project's `tsconfig.json` is.
+- `parserOptions.project` tells our parser that each source file's type information settings are described by the closest `tsconfig.json`
```
Comment (Josh-Cena) [inline docs/linting/Typed_Linting.mdx:37]:
```suggestion
- `parserOptions.project` tells our parser that each source file's type information settings are described by the closest `tsconfig.json`.
```

This is specifically for the value `true`. Maybe `parserOptions.project: true`? The notation looks horrible, though.

**Review by bradzacher — DISMISSED** [review]

File: docs/linting/Typed_Linting.mdx [hunk 6]
```diff
@@ -26,10 +26,15 @@ module.exports = {
...
+:::caution
+Your `.eslintrc.cjs` file may start receiving a parsing error about type information.
+See [our TSConfig inclusion FAQ](./Troubleshooting.mdx#i-get-errors-telling-me-eslint-was-configured-to-run--however-that-tsconfig-does-not--none-of-those-tsconfigs-include-this-file).
+:::
+
 In more detail:
 
 - `plugin:@typescript-eslint/recommended-requiring-type-checking` is another [recommended configuration](./CONFIGURATIONS.mdx) we provide. This one contains recommended rules that additionally require type information.
-- `parserOptions.project` tells our parser the relative path where your project's `tsconfig.json` is.
+- `parserOptions.project` tells our parser that each source file's type information settings are described by the closest `tsconfig.json`.
```
Comment (Josh-Cena) [inline docs/linting/Typed_Linting.mdx:37]: See https://github.com/typescript-eslint/typescript-eslint/pull/6919#discussion_r1169004478: should we make clear that we are talking about the value `true` here?

Comment (JoshuaKGoldberg) [reply docs/linting/Typed_Linting.mdx]: Oop, forgot to push - sorry!

**Review by Josh-Cena — APPROVED** [review]
Looks great to me 😀

File: docs/linting/Troubleshooting.mdx [hunk 7]
```diff
@@ -30,21 +30,67 @@ If you don't find an existing extension rule, or the extension rule doesn't work
...
+                'plugin:@typescript-eslint/recommended-requiring-type-checking',
+              ],
+              files: ['./**/*.{ts,tsx}'],
+            },
+          ],
+        };
+        ```
+
+      </details>
+      <details>
```
Comment (bradzacher) [inline docs/linting/Troubleshooting.mdx:65]: Maybe out this first so that the simpler config option is front and Center?

**Review by bradzacher — APPROVED** [review]

File: docs/linting/Troubleshooting.mdx (hunk 7 above)
Comment (JoshuaKGoldberg) [reply docs/linting/Troubleshooting.mdx:65]: Hmm interesting! Maybe once v6 lands we can? I personally lean towards side of putting additive changes in `overrides`. So maybe soon 😄

**Pull request merged**